coding-the-legislation/25_vectorial_computing.md

```diff
@@ -1,40 +1,183 @@
+    return condition * 200
+```
+
+> You should always use [Numpy function](https://docs.scipy.org/doc/numpy/reference/routines.math.html#sums-products-differences) such as [`where`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.where.html) and [`select`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.select.html) when they are relevant: logical operations using arithmetic operators should be used as last resort as they are not very readable.
+
+
+## Arithmetic operations
+
+Basic arithmetic operations such as `+` or `*` behave the same way on vectors than on numbers, you can thus use them in OpenFisca formulas. However, some operations must be adapted.
+
+| Scalar (won't work) | Vectorial alternative |
+|---------------------|-----------------------|
+| `min`               | `min_(x,y)`           |
+| `max`               | `max_(x,y)`           |
+| `round`             | `round_(x,y)`         |
+
+
+## Boolean operations
+
+| Scalar (won't work) | Vectorial alternative |
+|---------------------|-----------------------|
+| `not`               | `not_(x,y)`           |
+| `and`               | `x * y`               |
+| `or`             | `x + y`               |
 
 
-## Forbidden operations and alternatives
+## String concatenation
 
-Some classical control structures such as `if...else` or `switch`, and native python logical operators such as `or` and `not` **cannot** be used on a openfisca variable.
+The `+` operator, as well as formatted `%s` strings for concatenation should be replaced by a call to `concat(x, y)`.
 
-Here is a summary of the structures and operations you cannot use, and their licit alternatives:
 
+## Returning a single value for all entities
 
-| Forbidden             | Alternative                                                             |
-|-----------------------|-------------------------------------------------------------------------|
-| `if...else`, `switch` | `where`, `select`, `*` (See [Case disjunction](30_case_disjunction.md)) |
-| `not`                 | `not_(x)`                                                               |
-| `and`                 | `x * y`                                                                 |
-| `or`                  | `x + y`                                                                 |
-| `min`                 | `min_(x,y)`                                                             |
-| `max`                 | `max_(x,y)`                                                             |
-| `round`               | `round_(x,y)`                                                           |
-| `+` _(on strings)_    | `concat(x,y)`                                                           |
+If you really need to return the same value for every entity, the `np.ones` function allows you to create a vector of 1 of arbitrary length: `np.ones(persons.count) * some_parameter` will thus create a vector of the same length as the passed entities.
```

Comment:
Thanks a lot for that piece of information, I had almost added that note but thought there might be other cases in which it could be useful. If you also feel it should be a parameter, then I'll remove that note altogether. The doc should never encourage or even facilitate suboptimal implementations.